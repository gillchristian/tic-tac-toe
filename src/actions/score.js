import { createAction } from 'redux-actions'

export const SCORE_INCREMENT = 'score:increment'

// incrementScore :: X|O -> Redux.ActionObject
export const incrementScore = createAction(SCORE_INCREMENT)

