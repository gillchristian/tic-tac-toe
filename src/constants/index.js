export const O = '●'
export const X = '✖'

