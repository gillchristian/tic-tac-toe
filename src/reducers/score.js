import R from 'ramda'

import { X, O } from '../constants'
import { SCORE_INCREMENT } from '../actions/score'

export const SCORE_STATE = {
  [O]: 0,
  [X]: 0
}

export default function score(state = SCORE_STATE, action) {
  switch (action.type) {
    case SCORE_INCREMENT:
      return R.evolve({[action.payload]: R.inc}, state)
    default:
      return state
  }
}

// --- selectors ---
export const getScore = R.prop('score')

export const getO = R.prop(O)
export const getX = R.prop(X)

