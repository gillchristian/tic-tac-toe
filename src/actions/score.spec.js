import { incrementScore } from './score'

import { X, O } from '../constants'

describe('incrementScore', function () {
  it('creates an action with the provided mark as payload', () => {
    let actual = incrementScore(O)
    expect(actual).toMatchSnapshot()

    actual = incrementScore(X)
    expect(actual).toMatchSnapshot()

    actual = incrementScore('âœ“')
    expect(actual).toMatchSnapshot()
  })
})

