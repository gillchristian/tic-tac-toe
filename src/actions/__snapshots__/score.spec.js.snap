exports[`incrementScore creates an action with the provided mark as payload 1`] = `
Object {
  "payload": "●",
  "type": "score:increment",
}
`;

exports[`incrementScore creates an action with the provided mark as payload 2`] = `
Object {
  "payload": "✖",
  "type": "score:increment",
}
`;

exports[`incrementScore creates an action with the provided mark as payload 3`] = `
Object {
  "payload": "✓",
  "type": "score:increment",
}
`;
